client/components/state-dropdown/index.js

```diff
@@ -0,0 +1,44 @@
+import React, { PropTypes } from 'react';
+import Dropdown from 'components/dropdown';
+import TextField from 'components/text-field';
+import { translate as __ } from 'lib/mixins/i18n';
+
+const StateDropdown = ( props ) => {
+	const statesMap = ( props.countriesData[ props.countryCode ] || {} ).states;
+
+	if ( ! statesMap ) { // We don't have a list of states for this country
```

Comment:
If we don't have a list of states for a country, would it be safe to assume the country has no states and therefore we wouldn't even need this TextField? Or is that not a safe assumption.